### PR TITLE
fix(autoware_auto_tf2): modify build error in rolling

### PR DESCRIPTION
**Pull request by wep21**
## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Review by esteve — CHANGES_REQUESTED** [review]
Please apply the same solution as in https://github.com/autowarefoundation/autoware.universe/pull/720#discussion_r852040761

File: common/autoware_auto_tf2/CMakeLists.txt [hunk 1]
```diff
@@ -30,6 +30,10 @@ if(BUILD_TESTING)
   # Unit test
   ament_add_gtest(test_tf2_autoware_auto_msgs test/test_tf2_autoware_auto_msgs.cpp)
   autoware_set_compile_options(test_tf2_autoware_auto_msgs)
+  # workaround to allow deprecated header to build on both galactic and rolling
+  target_compile_options(test_tf2_autoware_auto_msgs PRIVATE
+    -Wno-error
```
Comment (esteve) [inline common/autoware_auto_tf2/CMakeLists.txt:35]: Please apply the same solution as in https://github.com/autowarefoundation/autoware.universe/pull/720#discussion_r852040761

**Review by esteve — CHANGES_REQUESTED** [review]

**Review by esteve — DISMISSED** [review]
@kenji-miyake there's a bunch of spelling mistakes that were not introduced by this branch, but that have popped up anyway, could you fix them? Otherwise I can deal with them.

**Comment by kenji-miyake** [comment]
@esteve Thank you for notifying me.
Okay, then I'll send another PR to fix them. Please review it later!

**Comment by kenji-miyake** [comment]
@wep21 Thank you for fixing typos, but I think it's better to split PRs.
So could you wait for #751?

**Comment by esteve** [comment]
@kenji-miyake sorry, I meant @wep21 :sweat_smile: , I guess I pinged you because I was reviewing #210

**Comment by kenji-miyake** [comment]
Oh, lol. :smile:
But yeah, it's alright.

**Comment by wep21** [comment]
@kenji-miyake 

> So could you wait for https://github.com/autowarefoundation/autoware.universe/pull/751?

I will rebase after #751 is merged.

**Comment by wep21** [comment]
@esteve @kenji-miyake I've finished. Could you review again?

**Review by kenji-miyake — APPROVED** [review]

**Comment by kenji-miyake** [comment]
@esteve It seems your comment https://github.com/autowarefoundation/autoware.universe/pull/718#discussion_r852042154 is resolved, I'll clear your review status. :bow:

**Pull request merged**